File: KotlinGeekTime/GeekTimeKotlin/src/com/zero/kotlinprimer/Chapter2/unit2/Main.java
package com.zero.kotlinprimer.Chapter2.unit2;

import kotlin.io.FilesKt;
import kotlin.text.Charsets;

import java.io.File;

public class Main {
    public static void main(String[] args){
        File file = new File("GeekTimeKotlin.iml");
        String content = FilesKt.readText(file, Charsets.UTF_8);
        System.out.println(content);
    }
}

File: KotlinGeekTime/GeekTimeKotlin/src/com/zero/kotlinprimer/Chapter2/unit2/Main.kt
package com.zero.kotlinprimer.Chapter2.unit2

import java.io.File

fun main(args: Array<String>){
    /*var file = File("GeekTimeKotlin.iml")
    println(file.readText())*/

    var dog = Dog()
    dog.printName(dog)
}

open class Animal
class Dog: Animal()

fun Animal.name() = "animal"
fun Dog.name() = "Dog"

fun Animal.printName(anim: Animal){
    println(anim.name())
}
